internal/client/client.go

```diff
@@ -52,26 +126,33 @@ func New[
 	Hasher runtime.Hasher[H],
 	N runtime.Number,
 	E runtime.Extrinsic,
+	Executor ExecutorT,
 	Header runtime.Header[N, H],
+	RA primitives_api.ConstructRuntimeApi[primitives_api.ApiExt[N, E, H, Hasher, statemachine.Backend[H, Hasher], any]],
```

Comment:
I think this will need to be revised in the future to a specific package level function in `client` package to utilize the `runtimeConstructor` to provide some other interface other than `ApiExt`.